doc/source/train/user-guides/checkpoints.rst

```diff
@@ -258,197 +167,163 @@ appropriately in distributed training.
-                scaling_config=ScalingConfig(num_workers=2),
-                resume_from_checkpoint=result.checkpoint,
-            )
-            result = trainer.fit()
+        .. literalinclude:: ../doc_code/checkpoints.py
+            :language: python
+            :start-after: __pytorch_restore_start__
+            :end-before: __pytorch_restore_end__
 
-            print(result.checkpoint.to_dict())
-            # {'epoch': 3, 'model_weights': OrderedDict([('bias', tensor([0.0902])), ('weight', tensor([[-0.1549, -0.0861,  0.4353, -0.4116]]))]), '_timestamp': 1656108265, '_preprocessor': None, '_current_checkpoint_id': 2}
 
     .. tab-item:: PyTorch Lightning
 
-        .. code-block:: python
-            :emphasize-lines: 11-17
-
-            from ray import train
-            from ray.train import Checkpoint, ScalingConfig
-            from ray.train.torch import TorchTrainer
-            from ray.train.lightning import RayTrainReportCallback
-            from os.path import join
-
-            def train_func_per_worker():
-                model = MyLightningModule(...)
-                datamodule = MyLightningDataModule(...)
-                trainer = pl.Trainer(
-                    ...
-                    callbacks=[RayTrainReportCallback()]
-                )
-
-                checkpoint = train.get_checkpoint()
-                if checkpoint:
-                    with checkpoint.as_directory() as ckpt_dir:
-                        ckpt_path = join(ckpt_dir, "checkpoint.ckpt")
-                        trainer.fit(model, datamodule=datamodule, ckpt_path=ckpt_path)
-                else:
-                    trainer.fit(model, datamodule=datamodule)
-
-            # Build a Ray Train Checkpoint
-            # Suppose we have a Lightning checkpoint under ./ckpt_dir/checkpoint.ckpt
-            checkpoint = Checkpoint.from_directory("./ckpt_dir/checkpoint.ckpt")
-
-            # Resume training from checkpoint file
-            ray_trainer = TorchTrainer(
-                train_func_per_worker,
-                scaling_config=ScalingConfig(num_workers=2),
-                resume_from_checkpoint=checkpoint,
-            )
-            result = ray_trainer.fit()
+        .. literalinclude:: ../doc_code/checkpoints.py
+            :language: python
+            :start-after: __lightning_restore_example_start__
+            :end-before: __lightning_restore_example_end__
+
+
+.. note::
+
+    In these examples, :meth:`Checkpoint.as_directory <ray.train.Checkpoint.as_directory>`
+    is used to view the checkpoint contents as a local directory.
+
+    *If the checkpoint points to a local directory*, this method just returns the
+    local directory path without making a copy.
+
+    *If the checkpoint points to a remote directory*, this method will download the
+    checkpoint to a local temporary directory and return the path to the temporary directory.
+
+    **If multiple processes on the same node call this method simultaneously,**
```

Comment:
Is this only guaranteed when calling as_directory inside the Ray Train training loop? Or it also works when you call it in different processes?

Reply:
works whenever you call in differnet processes in parallel